library/pkg/executor/executor.go

```diff
@@ -0,0 +1,359 @@
+// Package executor will execute an arbitrary command on linux in its own namespace and with restricted resources using cgroups
+// To use the executor package some all of the following prerequisites must be true
+// * The app using the executor must be run as root
+// * Cgroups v2 must be enabled on the host machine
+// * Cgroups v1 must be unmounted or empty, so it can be unmounted by the executor
+// * The host must cupport the io, memory, and cpu controllers in cgroups v2
+package executor
+
+import (
+	"errors"
+	"fmt"
+	"github.com/maczikasz/cu-thor/library/internal"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+type (
+	Executor struct {
+		cgroupsInfo map[string]string
+	}
+
+	//Config holds the configuration needed to setup the required cgroups in order to provide resource restrictions
+	//CpuPercent represent the allowed percentage of CPU time, it should be 0 < CpuPercent < 100
+	//MemoryBytes represent the accessible maximum memory in bytes
+	//WriteBytePerSec and ReadBytePerSec represent the allowed maximum bytes to be read from or written to, this will be applied to all disks
+	Config struct {
+		CpuPercent      int
+		MemoryBytes     int64
+		WriteBytePerSec int64
+		ReadBytePerSec  int64
+	}
+)
+
+var expectedControllers = []string{"io", "memory", "cpu"}
+
+const (
+	cgroupFolder     = "/sys/fs/cgroup"
+	defaultCPUPeriod = 100_000
+)
+
+// InitializeExecutorFromConfig will initialize a new executor by making sure that it has the neccesary cgroups setup to operate
+func InitializeExecutorFromConfig(config Config) (*Executor, error) {
+
+	if config.CpuPercent == 0 {
+		return nil, errors.New("invalid cpu percentage")
+	}
+
+	err := ensureCgroupsV2()
```

Comment:
This error is ignored.

Reply:
no it's not, just confusing because it's 2 lines below , will reorder the lines err handling should be after its manifestation